core/src/main/java/org/jdbi/v3/core/Handle.java

```diff
@@ -116,36 +115,51 @@ public Handle setStatementBuilder(StatementBuilder builder) {
      * rolled back.
      */
     @Override
+    @SuppressWarnings("PMD.DoNotThrowExceptionInFinally")
     public void close() {
-        if (!closed) {
-            boolean wasInTransaction = isInTransaction() && forceEndTransactions
-                    && config.get().get(Handles.class).isForceEndTransactions();
+        if (closed) {
+            return;
+        }
 
-            extensionMethod.remove();
-            config.remove();
+        boolean wasInTransaction = isInTransaction()
+            && forceEndTransactions
+            && config.get().get(Handles.class).isForceEndTransactions();
 
-            if (wasInTransaction) {
-                rollback();
-            }
+        extensionMethod.remove();
+        config.remove();
 
+        if (wasInTransaction) {
+            rollback();
+        }
+
+        Throwable suppressed = null;
+        try {
+            statementBuilder.close(getConnection());
+        } catch (Throwable t) {
+            suppressed = t;
+        } finally {
```

Comment:
I think putting a `finally` around this code block is actually redundant.

This block will always be reached.